src/modules/navigator/mission_block.cpp

```diff
@@ -339,22 +337,29 @@ MissionBlock::is_mission_item_reached_or_completed()
 			float alt_acc_rad_m = _navigator->get_altitude_acceptance_radius();
 
 			/* for vtol back transition calculate acceptance radius based on time and ground speed */
-			if (_mission_item.vtol_back_transition
-			    && _navigator->get_vstatus()->vehicle_type == vehicle_status_s::VEHICLE_TYPE_FIXED_WING) {
+			if (_mission_item.nav_cmd == NAV_CMD_VTOL_LAND || _mission_item.vtol_back_transition) {
 
-				float velocity = sqrtf(_navigator->get_local_position()->vx * _navigator->get_local_position()->vx +
-						       _navigator->get_local_position()->vy * _navigator->get_local_position()->vy);
+				if (_navigator->get_vstatus()->vehicle_type == vehicle_status_s::VEHICLE_TYPE_FIXED_WING) {
 
-				const float back_trans_dec = _navigator->get_vtol_back_trans_deceleration();
+					float velocity = sqrtf(_navigator->get_local_position()->vx * _navigator->get_local_position()->vx +
+							       _navigator->get_local_position()->vy * _navigator->get_local_position()->vy);
 
-				if (back_trans_dec > FLT_EPSILON && velocity > FLT_EPSILON) {
-					acceptance_radius = (velocity / back_trans_dec / 2) * velocity;
+					const float back_trans_dec = _navigator->get_vtol_back_trans_deceleration();
 
-				}
+					if (back_trans_dec > FLT_EPSILON && velocity > FLT_EPSILON) {
+						acceptance_radius = (velocity / back_trans_dec / 2) * velocity;
+
+					}
 
-				// do not care for altitude when approaching the backtransition point. Not accepting the waypoint causes
-				// the vehicle to perform a sharp turn after passing the land waypoint and this causes worse unexected behavior
-				alt_acc_rad_m = INFINITY;
+					// do not care for altitude when approaching the backtransition point. Not accepting the waypoint causes
+					// the vehicle to perform a sharp turn after passing the land waypoint and this causes worse unexected behavior
+					alt_acc_rad_m = INFINITY;
+
+				} else { //rotary wing
+					if (_mission_item.nav_cmd == NAV_CMD_VTOL_LAND) {
```

Comment:
What about a normal LAND command? Eg together with a backtransition waypoint?

Reply:
@sfuhrer That is handled by `vtol_back_transition` flag.
But it's good that you bring it up. I actually tested it and I find this worflow highly confusing. I did not even understand from the way it's planned in QGC where the actual backtransition is going to happen. Also the mavlink specs don't actually give any helpful information how to use this command. In the end, that's the reason why VTOL_LAND was invented but we still should increase clarity for this option.

Reply:
okay, if the behavior isn't changed for the separate transition and land items workflow it's all good for now.